### PR TITLE
examples changes for CICD

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
[examples changes for CICD]

### Issues Resolved
[List any issues this PR will resolve. e.g., Fixes #01]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/appliance_snmp_v3_trap_destinations.go [hunk 1]
```diff
@@ -12,24 +10,51 @@ func main() {
...
+
+	// Crete user
 
+	snmpv3User := ov.SNMPv3User{
+		UserName:                 "user0",
+		SecurityLevel:            "Authentication and privacy",
+		AuthenticationProtocol:   "SHA1",
+		AuthenticationPassphrase: "authPass",
+		PrivacyProtocol:          "AES-128",
+		PrivacyPassphrase:        "1234567812345678",
```
Comment (alisha-k-kalladassery) [inline examples/appliance_snmp_v3_trap_destinations.go:34]: Are we not taking these inputs through config.json?

File: examples/id_pools_ipv4_range.go [hunk 2]
```diff
@@ -14,40 +15,72 @@ import (
 func main() {
 	var (
 		ClientOV *ov.OVClient
-		id       = "ebb4dac4-6f9f-48b4-b356-fd371a026899"
-		eg_name  = "EG_For_IPV4Range"
+		//id       = "ebb4dac4-6f9f-48b4-b356-fd371a026899"
```
Comment (alisha-k-kalladassery) [inline examples/id_pools_ipv4_range.go:18]: is this a needed comment?

File: examples/id_pools_ipv4_range.go [hunk 3]
```diff
@@ -14,40 +15,72 @@ import (
...
+		config.OVCred.IfMatch)
+
+	//Create subnet
+	// Creates an IPv4 Subnet
+	subnet := ov.Ipv4Subnet{
+		// Name:       "SubnetGO",
+		// NetworkId:  "172.16.0.0",
+		// SubnetMask: "255.255.0.0",
+		// Gateway:    "172.16.0.1",
+		// Domain:     "Golang.com",
```
Comment (alisha-k-kalladassery) [inline examples/id_pools_ipv4_range.go:42]: Also this?

File: examples/id_pools_ipv4_range.go [hunk 4]
```diff
@@ -96,8 +147,10 @@ func main() {
 	// Allocates a set of IDs from an IPv4 range.
 	// A set of IDs can be allocated through count parameter also.
 	idlist := new([]utils.Nstring)
-	*idlist = append(*idlist, utils.NewNstring("<ip_address>"))
-	*idlist = append(*idlist, utils.NewNstring("<ip_address>"))
+	*idlist = append(*idlist, config.IdPoolsIpv4SubnetRange.IdList[0])
+	*idlist = append(*idlist, config.IdPoolsIpv4SubnetRange.IdList[1])
+	// *idlist = append(*idlist, utils.NewNstring("172.16.0.44"))
+	// *idlist = append(*idlist, utils.NewNstring("172.16.0.49"))
```
Comment (alisha-k-kalladassery) [inline examples/id_pools_ipv4_range.go:153]: comments here

File: examples/id_pools_ipv4_range.go [hunk 5]
```diff
@@ -170,4 +221,109 @@ func main() {
 		fmt.Println("#---Deleted Ipv4Range successfully---#")
 	}
 
+	err = ovc.DeleteIpv4Subnet(subnet_id)
+	if err != nil {
+		panic(err)
+	} else {
+		fmt.Println("#---Deleted Ipv4Range successfully---#")
+	}
+	//**************** Create mgmt network for automation ******************************//
```
Comment (alisha-k-kalladassery) [inline examples/id_pools_ipv4_range.go:230]: this needs to be uncommented right?

File: examples/id_pools_ipv4_subnets.go [hunk 6]
```diff
@@ -2,34 +2,41 @@ package main
...
-		Domain:     "Terraform.com",
+		Name:       config.IdPoolsIpv4Subnet.SubnetName,
+		NetworkId:  config.IdPoolsIpv4Subnet.NetworkId,
+		SubnetMask: config.IdPoolsIpv4Subnet.SubnetMask,
+		Gateway:    config.IdPoolsIpv4Subnet.Gateway,
+		Domain:     config.IdPoolsIpv4Subnet.Domain,
 	}
+	// 	Name:       config.IdPoolsIpv4Subnet.SubnetName,
+	// 	SubnetMask: "255.255.0.0",
+	// 	Gateway:    "172.16.0.1",
```
Comment (alisha-k-kalladassery) [inline examples/id_pools_ipv4_subnets.go:37]: comments here

File: examples/logical_enclosures.go [hunk 7]
```diff
@@ -2,45 +2,73 @@ package main
...
+		fmt.Println(config_err)
+	}
 	var (
 		ClientOV            *ov.OVClient
 		logical_enclosure   = "TestLE"
 		logical_enclosure_1 = "TestLE-Renamed"
 		scope_name          = "Auto-Scope"
-		eg_name             = "EG"
+		eg_name             = config.EgName
 		firmware_baseline   = utils.NewNstring("/rest/firmware-drivers/Synergy_Custom_SPP_2021_02_01_Z7550-97110")
```
Comment (alisha-k-kalladassery) [inline examples/logical_enclosures.go:24]: should we pass this in config?

File: examples/logical_enclosures.go [hunk 8]
```diff
@@ -2,45 +2,73 @@ package main
...
+	for i := 0; i < len(enc_list_uris); i++ {
+		*enclosureUris = append(*enclosureUris, utils.NewNstring(enc_list_uris[i]))
+
+	}
+
+	fmt.Println(*enclosureUris)
+
+	// fmt.Println("#................... Create Logical Enclosure ...............#")
+	// enclosureUris := new([]utils.Nstring)
+	// *enclosureUris = append(*enclosureUris, utils.NewNstring("/rest/enclosures/0000000000A66101"))
```
Comment (alisha-k-kalladassery) [inline examples/logical_enclosures.go:62]: Are these needed?

File: examples/logical_interconnect_groups.go [hunk 9]
```diff
@@ -157,9 +159,27 @@ func main() {
...
+		fmt.Println(err)
+	}
+	fcNetwork, err := ovc.GetFCNetworkByName(config.FcNetworkName)
+	if err != nil {
+		fmt.Println(err)
+	}
+	networkUris := []utils.Nstring{EthNetworkMgmt.URI}
+	networkUris2 := []utils.Nstring{EthNetworkIscsi.URI}
+	networkUris3 := []utils.Nstring{fcNetwork.URI}
+	// networkUris := []utils.Nstring{"/rest/ethernet-networks/bf5a6091-ef32-4983-90b7-d5a3848c7274"}
```
Comment (alisha-k-kalladassery) [inline examples/logical_interconnect_groups.go:180]: Are these needed?

File: examples/server_hardware.go [hunk 10]
```diff
@@ -30,9 +31,9 @@ func main() {
 	fmt.Println("-----------------------------")
 	fmt.Println("Add Single Rack server to the appliance:")
 	rackServer := ov.ServerHardware{
-		Hostname:           "<serverIp>",
-		Username:           "<username>",
-		Password:           "<password>",
+		Hostname:           "172.18.41.1",
+		Username:           "dcs",
+		Password:           "dcs",
```
Comment (alisha-k-kalladassery) [inline examples/server_hardware.go:36]: server hw ips in server_hardware.go

Comment (nabhajit-ray) [reply examples/server_hardware.go]: fixed

File: examples/logical_interconnect_groups.go (hunk 9 above)
Comment (nabhajit-ray) [reply examples/logical_interconnect_groups.go]: fixed

File: examples/id_pools_ipv4_range.go (hunk 3 above)
Comment (nabhajit-ray) [reply examples/id_pools_ipv4_range.go]: fixed

File: examples/id_pools_ipv4_range.go (hunk 2 above)
Comment (nabhajit-ray) [reply examples/id_pools_ipv4_range.go]: fixed

File: examples/id_pools_ipv4_range.go (hunk 4 above)
Comment (nabhajit-ray) [reply examples/id_pools_ipv4_range.go]: removed

File: .github/workflows/createtagandrelease.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    if: github.event.pull_request.merged && startsWith(github.head_ref, 'PR') && startsWith(github.base_ref, 'master')
+    steps:
+      - uses: actions/checkout@v2
+      - name: Bump version and push tag
+        id: tag_and_prepare_release
+        uses: moble/github-tag-action@main
+        env:
+          GITHUB_TOKEN:  ${{ secrets.CD_TOKEN }}
+          WITH_V: true
+          CUSTOM_TAG: "v7.2.0"
```
Comment (AvijitOS) [inline .github/workflows/createtagandrelease.yml:20]: custom tag should be dynamic right

File: examples/appliance_snmp_v1_trap_destinations.go [hunk 12]
```diff
@@ -2,24 +2,39 @@ package main
...
-	"os"
-	"strconv"
 )
 
 func main() {
 	var (
 		ClientOV *ov.OVClient
 	)
-	apiversion, _ := strconv.Atoi(os.Getenv("ONEVIEW_APIVERSION"))
+	/*
```
Comment (AvijitOS) [inline examples/appliance_snmp_v1_trap_destinations.go:13]: need to remove this commented section

File: examples/appliance_snmp_v1_trap_destinations.go [hunk 13]
```diff
@@ -2,24 +2,39 @@ package main
...
+		config.OVCred.UserName,
+		config.OVCred.Password,
+		config.OVCred.Domain,
+		config.OVCred.Endpoint,
+		config.OVCred.SslVerify,
+		config.OVCred.ApiVersion,
+		config.OVCred.IfMatch)
 
 	snmpUser := ov.SNMPv1Trap{
 		Destination:     "192.0.1.14",
```
Comment (AvijitOS) [inline examples/appliance_snmp_v1_trap_destinations.go:40]: snmp trap destination ip should come from config?

**Review by AvijitOS — COMMENTED** [review]
can u send the list of examples will be supported in golang CD currently

File: examples/appliance_snmp_v1_trap_destinations.go (hunk 13 above)
Comment (nabhajit-ray) [reply examples/appliance_snmp_v1_trap_destinations.go:40]: Yeah that is random ip

File: .github/workflows/createtagandrelease.yml (hunk 11 above)
Comment (nabhajit-ray) [reply .github/workflows/createtagandrelease.yml:20]: custom tag supposed to get change during release workflow

File: examples/logical_enclosures.go (hunk 7 above)
Comment (nabhajit-ray) [reply examples/logical_enclosures.go:24]: EG name is picked up from config as enclosure group will create it

File: examples/id_pools_ipv4_subnets.go (hunk 6 above)
Comment (nabhajit-ray) [reply examples/id_pools_ipv4_subnets.go]: removed

File: examples/id_pools_ipv4_range.go (hunk 5 above)
Comment (nabhajit-ray) [reply examples/id_pools_ipv4_range.go]: Commented part not needed and removed

File: examples/logical_enclosures.go (hunk 7 above)
Comment (nabhajit-ray) [reply examples/logical_enclosures.go:24]: Will remove the firmware baseline

File: examples/logical_enclosures.go (hunk 8 above)
Comment (nabhajit-ray) [reply examples/logical_enclosures.go]: Removed

**Review by alisha-k-kalladassery — APPROVED** [review]

**Review by AvijitOS — APPROVED** [review]

**Pull request merged**